### PR TITLE
First attempt to implement module detection

### LINKED ISSUE #7: Ensure Polymer 3 Modulizer compatibility
See “Generic steps” at vaadin/components-team-tasks#326

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #7

Depends on vaadin/vaadin-usage-statistics#11

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-development-mode-detector/8)

### CODE REVIEW

File: vaadin-development-mode-detector.html [hunk 1]
```diff
@@ -67,6 +92,23 @@
         }
       }, function() {}, true);
     }
+
+    const loadAndRun = function(id, optionalArgument) {
+      let path = preparePath(id, ".js");
+      let src = document.createElement("script");
+      src.setAttribute("src", path);
+      src.async = true;
+      src.onreadystatechange = src.onload = function() {
```
Comment (limonte) [inline vaadin-development-mode-detector.html:101]: From https://stackoverflow.com/a/16231055/1331425

> You should set the `src` attribute **after** the `onload` event

> You should also append the script to the DOM **before** attaching the `onload` event

**Comment by web-padawan** [comment]
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-development-mode-detector.html, line 101 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-development-mode-detector/8#-L7y11v-D8LlKidtIwaz-r1-101:-L7yKGw98e7m1THjeear:bnh6ch6) ([raw file](https://github.com/vaadin/vaadin-development-mode-detector/blob/b5458f29c7e38515f3040fb690c46e4a09afa23b/vaadin-development-mode-detector.html#L101)):*
<details><summary><i>Previously, limonte (Limon Monte) wrote…</i></summary><blockquote>

From https://stackoverflow.com/a/16231055/1331425

> You should set the `src` attribute **after** the `onload` event

> You should also append the script to the DOM **before** attaching the `onload` event
</blockquote></details>

That comment seems to be related to the jQuery stuff, actually.

See the working code example at https://github.com/LasaleFamine/polymer-lib-loader/blob/master/lib-loader.es6.js#L70

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-development-mode-detector/8)*

**Comment by tomivirkki** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-development-mode-detector/8#-:-L7yacAHbOAr2m-0iLdQ:bnfp4nl)*

**Comment by limonte** [comment]
:lgtm: 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-development-mode-detector.html, line 101 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-development-mode-detector/8#-L7y11v-D8LlKidtIwaz-r1-101:-L7ydY_PG4MwCGx5cdBa:b3csspz) ([raw file](https://github.com/vaadin/vaadin-development-mode-detector/blob/b5458f29c7e38515f3040fb690c46e4a09afa23b/vaadin-development-mode-detector.html#L101)):*
<details><summary><i>Previously, web-padawan (Sergey Kulikov) wrote…</i></summary><blockquote>

That comment seems to be related to the jQuery stuff, actually.

See the working code example at https://github.com/LasaleFamine/polymer-lib-loader/blob/master/lib-loader.es6.js#L70

</blockquote></details>

true that, verified in local sandbox.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-development-mode-detector/8#-:-L7ydcQmzCeM67Rb-Fia:bg7igkv)*

**Pull request merged**